### PR TITLE
interface: instability of the interface language

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Closes rero/sonar#741.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Why are you opening this PR?

- stabilizing the interface language

## How to test?

- Change the language on different pages of the site and then navigate.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/sonar/src/app/app-translate-loader.ts [hunk 1]
```diff
@@ -59,11 +58,18 @@ export class AppTranslateLoader extends NgCoreTranslateLoader {
...
           ).bibCode;
           [1, 2, 3].forEach((id: number) => {
-            const label = user.organisation['documentsCustomField' + id].label;
-            if (label) {
-              const entry = label.find((lab: any) => lab.language === bibLanguage);
-              if (entry) {
-                translation['Custom field ' + id] = entry.value;
+            const key = 'documentsCustomField' + id;
+            if (key in user.organisation) {
+              if ('label' in user.organisation[key]) {
```
Comment (jma) [inline projects/sonar/src/app/app-translate-loader.ts:63]: two `if` can be merged with an `and`.

File: projects/sonar/src/app/app-translate-loader.ts [hunk 2]
```diff
@@ -59,11 +58,18 @@ export class AppTranslateLoader extends NgCoreTranslateLoader {
...
+                const label = user.organisation[key].label;
+                const entry = label.find((lab: any) => lab.language === bibLanguage);
+                if (entry) {
+                  translation['Custom field ' + id] = entry.value;
+                } else {
+                  // If we do not have a value for the selected language,
+                  // we take the first value in the array.
+                  translation['Custom field ' + id] = label[0].value;
```
Comment (jma) [inline projects/sonar/src/app/app-translate-loader.ts:71]: can be replaced by `Custom field ${id}`

File: projects/sonar/src/app/_layout/admin/admin.component.ts [hunk 3]
```diff
@@ -112,5 +114,17 @@ export class AdminComponent implements OnInit, OnDestroy {
...
+  }
+
+  /**
+   * Change language on frontend flask.
+   *
+   * @param languageCode 2 digit language code.
+   */
+  private _changeFlaskLanguage(languageCode: string): void {
+    this._httpClient
+      .get(`/lang/${languageCode}`, { responseType: 'text' })
```
Comment (jma) [inline projects/sonar/src/app/_layout/admin/admin.component.ts:127]: A `GET` method seems not really clean, but maybe it is related the to backend API.

**Review by jma — APPROVED** [review]

File: projects/sonar/src/app/_layout/admin/admin.component.ts (hunk 3 above)
Comment (Garfield-fr) [reply projects/sonar/src/app/_layout/admin/admin.component.ts:127]: Exactly. It depends on the backend

**Pull request merged**